src/webgpu/shader/execution/robust_access_vertex.spec.ts

```diff
@@ -0,0 +1,40 @@
+export const description = `
+Test vertex attributes behave correctly (no crash / data leak) when accessed out of bounds
+`;
+
+/*
+Test coverage:
+
+The following will be parameterized (all combinations tested):
+
+1) Draw call indexed? (false / true)
+  - Run the draw call using an index buffer
+
+2) Draw call indirect? (false / true)
+  - Run the draw call using an indirect buffer
+
+3) Draw call parameter (vertexCount, firstVertex, indexCount, firstIndex, baseVertex, instanceCount,
+  firstInstance)
+  - The parameter which will go out of bounds. Filtered depending on if the draw call is indexed.
+
+4) Attribute type (float, vec2, vec3, vec4, mat2, mat3, mat4)
+  - The input attribute type in the vertex shader
+
+5) Error scale (1, 100, 100000000)
+  - Offset to add to the correct draw call parameter
+
+6) Additional vertex buffer (false / true)
```

Comment:
Maybe we could try varying the number of vertex buffers, or the one that goes OOB to make sure programmable vertex pulling is implemented correctly for multiple vertex buffers?

Reply:
Also consider one or multiple vertex attributes per vertex buffer since if the implementation does vertex pulling it'll need to load more than one attribute from a single buffer.

Reply:
Added some text to say we'll check varied number of vertex buffers (+2 / +4 more buffers) and that the test will bind multiple attributes to a buffer.